File: test-suite/src/main/java/example/micronaut/aiservice/tools/LegalDocumentTools.java
package example.micronaut.aiservice.tools;

import dev.langchain4j.agent.tool.Tool;
import jakarta.inject.Singleton;

import java.time.LocalDate;

@Singleton // <1>
public class LegalDocumentTools {
    @Tool("Returns the last time the PRIVACY document was updated") // <2>
    public LocalDate lastUpdatePrivacy() {
        return LocalDate.of(2013, 3, 9); // <3>
    }
}

File: test-suite-python/src/main/python/example/micronaut/aiservice/tools/LegalDocumentTools.py
from dev.langchain4j.agent.tool import Tool
from jakarta.inject import Singleton
from java.time import LocalDate


@Singleton  # <1>
class LegalDocumentTools:
    @Tool("Returns the last time the PRIVACY document was updated")  # <2>
    def last_update_privacy(self) -> LocalDate:
        return LocalDate.of(2013, 3, 9)  # <3>
